OpenEdXMobile/src/main/java/org/edx/mobile/view/MyCoursesListFragment.java

```diff
@@ -299,9 +375,101 @@ public boolean onOptionsItemSelected(MenuItem item) {
                 environment.getRouter().showFindCourses(getContext());
                 return true;
             }
+            case R.id.menu_item_account: {
+                environment.getRouter().showAccountActivity(getActivity());
+                //TODO: remove following code block after testing, (once LEARNER-3251 is done)
+//                if (toolbarCallbacks.getSearchView().getVisibility() == View.VISIBLE) {
+//                    hideSearchBar();
+//                } else {
+//                    showSearchBar();
+//                }
+                return true;
+            }
             default: {
                 return super.onOptionsItemSelected(item);
             }
         }
     }
+
+    private void hideSearchBar() {
+        toolbarCallbacks.getSearchView().setVisibility(View.GONE);
+        toolbarCallbacks.getTitleView().setVisibility(View.VISIBLE);
+    }
+
+    //TODO: This function has to update once LEARNER-3251 is done
+    private void showSearchBar() {
+        final SearchView searchView = toolbarCallbacks.getSearchView();
+        searchView.setVisibility(View.VISIBLE);
+        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
+            @Override
+            public boolean onQueryTextSubmit(String query) {
+                searchView.onActionViewCollapsed();
+                return true;
+            }
+
+            @Override
+            public boolean onQueryTextChange(String newText) {
+                return false;
+            }
+        });
+        searchView.setOnQueryTextFocusChangeListener(new View.OnFocusChangeListener() {
+            @Override
+            public void onFocusChange(View view, boolean queryTextFocused) {
+                final TextView titleView = toolbarCallbacks.getTitleView();
+                if (queryTextFocused) {
+                    titleView.setVisibility(View.GONE);
+                } else {
+                    titleView.setVisibility(View.VISIBLE);
+                    searchView.onActionViewCollapsed();
+                }
+            }
+        });
+    }
+
+    private void loadProfileImage(@NonNull ProfileImage profileImage, @NonNull ImageView imageView) {
+        if (profileImage.hasImage()) {
+            Glide.with(this)
+                    .load(profileImage.getImageUrlMedium())
+                    .into(imageView);
+        } else {
+            Glide.with(this)
+                    .load(R.drawable.profile_photo_placeholder)
+                    .into(imageView);
+        }
+    }
+
+    @SuppressWarnings("unused")
+    public void onEventMainThread(@NonNull ProfilePhotoUpdatedEvent event) {
+        if (!environment.getConfig().isTabsLayoutEnabled() || !environment.getConfig().isUserProfilesEnabled()) {
+            return;
+        }
+        final ImageView profileImage = toolbarCallbacks.getProfileView();
```

Comment:
Starting from here, the code in this function is copied at many places wherever `ProfilePhotoUpdatedEvent` is being consumed. Would be better to move this code into a Util class and standardize its usage in all other classes as well.